src/pedal_handler.cpp

```diff
@@ -44,97 +57,113 @@ int PedalHandler::calculate_torque(int16_t &motor_speed, int &max_torque, bool r
     return calculated_torque;
 }
 
+int16_t PedalHandler::calculate_regen(int16_t &motor_speed, int16_t max_regen_torque)
+{
+    int16_t calculated_regen_torque = 0;
+    // regen_command = ;
+    // 40191 = -10nm
+    // 10101 = -100nm
+    uint16_t calculated_torque2 = this->bse1.getTravelRatio() * (uint16_t)(-(REGEN_NM * 10));
+    calculated_regen_torque = static_cast<int>(calculated_torque2);
+    // calculated_torque = 40191;
+    Serial.print("regen torque: ");
+    Serial.println(calculated_regen_torque);
+    return calculated_regen_torque;
+}
 // returns true if the brake pedal is active
 bool PedalHandler::read_pedal_values()
 {
     /* Filter ADC readings */
 
+    // exponential smoothing https://chat.openai.com/share/cf98f2ea-d87c-4d25-a365-e398ffebf968
+    // TODO - evaluate this ALPHA value and if maybe we should decrease it
     accel1_ =
         ALPHA * accel1_ + (1 - ALPHA) * pedal_ADC.read_adc(ADC_ACCEL_1_CHANNEL);
     accel2_ =
         ALPHA * accel2_ + (1 - ALPHA) * pedal_ADC.read_adc(ADC_ACCEL_2_CHANNEL);
     brake1_ =
         ALPHA * brake1_ + (1 - ALPHA) * pedal_ADC.read_adc(ADC_BRAKE_1_CHANNEL);
-    
-    steering_angle_ = pedal_ADC.read_adc(3);
 
-    if(debugPrint.check()){
-        #ifdef DEBUG
-        Serial.print("ADC1 :");
+    steering_angle_ =
+        ALPHA * steering_angle_ + (1 - ALPHA) * pedal_ADC.read_adc(ADC_HALL_CHANNEL);
+
+    if (debugPrint.check())
+    {
+#if DEBUG
+        Serial.print("APPS1 :");
         Serial.println(accel1_);
-        Serial.print("ADC2 :");
+        Serial.print("APPS2 :");
         Serial.println(accel2_);
         Serial.print("BRAKE :");
         Serial.println(brake1_);
-        #endif
+        Serial.print("STEERING_ANGLE :");
+        Serial.println(steering_angle_);
+#endif
     }
 
-    // This is the code to print raw ADC readings vs the filtered one
     VCUPedalReadings.set_accelerator_pedal_1(accel1_);
     VCUPedalReadings.set_accelerator_pedal_2(accel2_);
     VCUPedalReadings.set_brake_transducer_1(brake1_);
     VCUPedalReadings.set_brake_transducer_2(steering_angle_);
-    CAN_message_t tx_msg;
-    CAN_message_t tx_msg2;
-    // Send Main Control Unit pedal reading message
-    VCUPedalReadings.write(tx_msg.buf);
-    tx_msg.id = ID_VCU_PEDAL_READINGS;
-    tx_msg.len = sizeof(VCUPedalReadings);
-    tx_msg2.id = ID_VCU_WS_READINGS;
-    tx_msg.len = 8;
-    uint32_t rpm_wsfl = (int)(current_rpm * 100);
-    uint32_t rpm_wsfr = (int)(current_rpm2 * 100);
-    memcpy(&tx_msg2.buf[0], &rpm_wsfl, sizeof(rpm_wsfl));
-    memcpy(&tx_msg2.buf[4], &rpm_wsfr, sizeof(rpm_wsfr));
-
-    // write out the actual accel command over CAN
-    if (pedal_out->check())
-    {
-        WriteToDaqCAN(tx_msg);
-        WriteToDaqCAN(tx_msg2);
-    }
     // only uses front brake pedal
     brake_is_active_ = (brake1_ >= BRAKE_ACTIVE);
     return brake_is_active_;
 }
 
+void PedalHandler::send_readings()
+{
+    if (pedal_out->check())
```

Comment:
ask stakeholders what frequency they want this to be reported

Reply:
off the hip 100ms? idk tho